<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  
  <script>
    // // 请求一: 
    // let isResult1 = false;
    // let isResult2 = false;
    // $.ajax({
    //   url: '',
    //   success: function () {
    //     console.log('结果1');
    //     isResult1 = true;
    //     handleresult();
    //   }
    // });
    // // 请求二: 
    // $.ajax({
    //   url: '',
    //   success: function () {
    //     console.log('结果2');
    //     isResult2 = true;
    //     handleresult();
    //   }
    // });

    // function handleresult() {
    //   if (isResult1 && isResult2) {

    //   }
    // }

    
    // 网络请求: aaa -> 自己处理(10行)
    // 处理: aaa111 -> 自己处理(10行)
    // 处理: aaa111222 -> 自己处理
    // new Promise((resolve, reject) => {
    //   setTimeout(() => {
    //     resolve('aaa');
    //   }, 1000);
    // }).then(value => {
    //   // 1. 自己处理10行代码
    //   console.log(value, '第一层的10行代码');
      
    //   // 2. 对结果进行第一次处理
    //   return new Promise((resolve, reject) => {
    //     // resolve(value + '111');
    //     reject('err');
    //   });
    // }).then(value => {
    //   console.log(value, '第二层的10行处理代码');

    //   return new Promise(resolve => {
    //     resolve(value + '222');
    //   }); 
    // }).then(value => {
    //   console.log(value, '第三层处理');
    // }).catch(err => {
    //   console.log(err);
    // });

    // new Promise(resolve => resolve(结果)); 简写
    // new Promise((resolve, reject) => {
    //   setTimeout(() => {
    //     resolve('aaa');
    //   }, 1000);
    // }).then(value => {
    //   // 1. 自己处理10行代码
    //   console.log(value, '第一层的10行代码');
      
    //   // 2. 对结果进行第一次处理
    //   return Promise.reject('err message');
    // }).then(value => {
    //   console.log(value, '第二层的10行处理代码');

    //   return Promise.resolve(value + '222'); 
    // }).then(value => {
    //   console.log(value, '第三层处理');
    // }).catch(err => {
    //   console.log(err);
    // });

    // 省略Promise.resolve
    new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve('aaa');
      }, 1000);
    }).then(value => {
      // 1. 自己处理10行代码
      console.log(value, '第一层的10行代码');
      
      // 2. 对结果进行第一次处理
      throw 'error message';
    }).then(value => {
      console.log(value, '第二层的10行处理代码');

      return value + '222'; 
    }).then(value => {
      console.log(value, '第三层处理');
    }).catch(err => {
      console.log(err);
    });
  </script>
</body>
</html>